include/amber/amber.h

```diff
@@ -52,6 +52,12 @@ struct BufferInfo {
   std::vector<Value> values;
 };
 
+/// Delegate class to hook functions around graphics API commands
+class Delegate {
+ public:
+  virtual void log(std::string command) = 0;
+};
```

Comment:
See below, I think we should add a:
```
  bool LogGraphicsCalls() const { ... }
```

And a setting so the sample app can set that value into the delegate. (Which also means we need a virtual destructor as well)